xdsl/dialects/arith.py

```diff
@@ -775,6 +794,33 @@ class ExtSIOp(IRDLOperation):
     def __init__(self, op: SSAValue | Operation, target_type: IntegerType):
         return super().__init__(operands=[op], result_types=[target_type])
 
+    def verify_(self) -> None:
+        assert isinstance(self.input.type, IntegerType)
+        assert isinstance(self.result.type, IntegerType)
```

Comment:
This is already encoded in the field definitions, right?

Reply:
This is for pyright reasons